core/src/main/java/hudson/slaves/ChannelPinger.java

```diff
@@ -87,25 +107,53 @@ public void install(Channel channel) {
 
         // set up ping from both directions, so that in case of a router dropping a connection,
         // both sides can notice it and take compensation actions.
-        setUpPingForChannel(channel, pingInterval);
+        setUpPingForChannel(channel, pingTimeoutSeconds, pingIntervalSeconds);
     }
 
-    private static class SetUpRemotePing extends MasterToSlaveCallable<Void, IOException> {
-        private static final long serialVersionUID = -2702219700841759872L;
-        private int pingInterval;
-        public SetUpRemotePing(int pingInterval) {
-            this.pingInterval = pingInterval;
+    static class SetUpRemotePing extends MasterToSlaveCallable<Void, IOException> {
+        private final int pingTimeoutSeconds;
```

Comment:
There are issues with differences in JVM versions between master and slave. I have customer issues where not having a serialVersionUID defined has caused Jenkins to blow up (thankfully all of these cases were in plugins). Just because some instances of callables in Core have been lucky so far does not mean it is a safe pattern to follow.

In addition, assuming that the Jenkins core implementation of Remoting (with its current contract of classloaders) is the only implementation is a risky assumption. We have an alternative implementation for Master-to-Master communication which (for security reasons) does not allow remote class loading and as such the serialVersionUID must match for any callables that are transferred over that Channel. While we are not in a position to release that implementation at the present time, I know it is on our roadmap to release it in the future. It would be better if the habit of serialVersionUID *alway* was better applied in Core so that when we do open source this code we can set an earlier lower-bound on the Master version that the code applies to.